.circleci/config.yml

```diff
@@ -723,6 +723,87 @@ jobs:
       - store_test_results:
           path: ./reports/junit
 
+  # -------------------------
+  #     JOBS: iOS E2E Tests
+  # -------------------------
+  test_e2e_ios:
+    executor: reactnativeios
+    parameters:
+      ruby_version:
+        default: "2.7.7"
+        description: The version of ruby that must be used
+        type: string
+    environment:
+      - PROJECT_NAME: "RNTesterE2E"
+    steps:
+      - checkout_code_with_cache
+      - run_yarn
+      - run:
+          name: Install appium
+          command: npm install appium@2.0.0-rc.4 -g
+      - run:
+          name: Install appium drivers
+          command: |
+            appium driver install uiautomator2
+            appium driver install xcuitest
+      - run:
+          name: Start Appium server
+          command: appium --base-path /wd/hub
+          background: true
+      - run:
+          name: Start Metro
+          command: |
+            cd packages/rn-tester
+            yarn start
+          background: true
+      - brew_install:
+          package: cmake
+      - run:
+          name: Check Appium server status
+          command: |
+            if ! nc -z 127.0.0.1 4723; then
+              echo Could not find Appium server
+              exit 1
+            fi
+      - attach_workspace:
+          at: .
+      - setup_ruby:
+          ruby_version: << parameters.ruby_version >>
+      - run: |
+          cd packages/rn-tester
+          bundle check || bundle install
+          bundle exec pod setup
+          bundle exec pod install --verbose
+      - run:
+          name: Boot iOS Simulator
+          command: source scripts/.tests.env && xcrun simctl boot "$IOS_DEVICE" || true
+      - run:
+          name: Configure Environment Variables
+          command: |
+            echo 'export PATH=/usr/local/opt/node@18/bin:$PATH' >> $BASH_ENV
+            source $BASH_ENV
+      - run:
+          name: Configure Watchman
+          command: echo "{}" > .watchmanconfig
+      - run:
+          name: Build app
+          command: |
+              xcodebuild build \
+                -workspace packages/rn-tester/RNTesterPods.xcworkspace \
+                -configuration Debug \
+                -scheme RNTester \
+                -sdk iphonesimulator \
+                -derivedDataPath /tmp/e2e/
+
+      - run:
+          name: Move app to correct directory
+          command: mv /tmp/e2e/Build/Products/Debug-iphonesimulator/RNTester.app packages/rn-tester-e2e/apps/
+      - run:
+          name: Run E2E tests
+          command: | 
+            cd packages/rn-tester-e2e 
+            yarn test-ios-e2e
```

Comment:
if the test-e2e produces some artifacts, we can think about saving them in CircleCI using the [`store_artifacts`](https://circleci.com/docs/artifacts/#uploading-artifacts) instruction